sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -9,20 +9,27 @@
 using System.Reflection;
 using NUnit.Framework;
 using NUnit.Framework.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Resources;
 
 namespace Azure.Core.TestFramework
 {
     [Category("Recorded")]
-    public abstract class RecordedTestBase : ClientTestBase
+    public abstract class RecordedTestBase<TEnvironment> : ClientTestBase
+    where TEnvironment : TestEnvironment, new()
     {
         protected RecordedTestSanitizer Sanitizer { get; set; }
 
         protected RecordMatcher Matcher { get; set; }
 
         public TestRecording Recording { get; private set; }
 
-        public RecordedTestMode Mode { get; set; }
+        private static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
 
+        protected ResourceGroupCleanupPolicy CleanupPolicy { get; set; }
+
```

Comment:
Should this be a little more then just the clean-up policy? It would be good if it could be a factory so it can be in charge of creating, tracking, and cleaning up the resource groups.

Reply:
I have not change the names of anything yet. I just wanted to get everything moved over (out of common/ManagementTestShared) first in a separate PR with minimal changes to remove ManagmentRecordTestBase, RecordedTestBase{TEnviroment} and moving ResourceGroupCleanupPolicy

Reply:
We are staying with parity with this PR